bug(analytics): HealthScore interface missing 'components' field causes TS2353 in CodeQualityDashboard

## Problem

`vue-tsc` reports a `TS2353` error in `CodeQualityDashboard.vue`:

```
src/components/analytics/CodeQualityDashboard.vue(736,9): error TS2353: Object literal may only specify known properties, and 'components' does not exist in type 'HealthScore | ...'
```

## Root Cause

The `HealthScore` interface (line 463) is missing a `components` field:

```ts
interface HealthScore {
  overall: number;
  grade: string;
  trend: number;
  breakdown: Record<string, number>;
  recommendations: string[];
  // 'components' is missing here
}
```

But `loadHealthScore()` (line 736) assigns:
```ts
healthScore.value = {
  ...
  components: data.components || { code_quality: 0, security: 0, performance: 0 },
  ...
}
```

## Fix

Add `components: Record<string, number>` to the `HealthScore` interface.

## Acceptance Criteria
- [ ] `vue-tsc --noEmit -p tsconfig.app.json` reports no TS2353 errors in `CodeQualityDashboard.vue`

--- a/autobot-frontend/src/components/analytics/CodeQualityDashboard.vue
+++ b/autobot-frontend/src/components/analytics/CodeQualityDashboard.vue
@@ -465,6 +465,7 @@ interface HealthScore {
   grade: string;
   trend: number;
   breakdown: Record<string, number>;
+  components: Record<string, number>;
   recommendations: string[];
 }
 
@@ -512,6 +513,7 @@ const healthScore = ref<HealthScore>({
   grade: 'C',
   trend: 0,
   breakdown: {},
+  components: {},
   recommendations: [],
 });
 

